### DISCUSSION TITLE
Machine learning primary and secondary machine

### DISCUSSION THREAD

**Discussion by Pompiedom** [immich-app/immich#5772] [General]
Hi, I would love to add a secondary machine-learning url when the first one isn't available so machine-learning can always happen.

Situation:
I run immich on a NAS which is fast enough to run immich but machine-learning through 200k photos isn't going fast. So I setup docker with machine-learning on a pc specially for this purpose but this machine isn't always on. 

Wish:
For this I would love to have a primary-secondary machine option. For example: immich checks (interval?) if the first url is available and if it is it picks that machine to run it's tasks and if it isn't it checks/picks the secondary one. This can even be a comma separated list until it has a machine to run its tasks.

**Comment by bo0tzz** [comment]
You can set up a reverse proxy or loadbalancer that Immich sends the ML requests to, which then forwards them to the different machines however you want.
